test/manifest.e2e-spec.ts

```diff
@@ -0,0 +1,382 @@
+        },
+      ],
+      startBlock: newBlock.number,
+      endBlock: newBlock.number,
+    });
+
+    jest
+      .spyOn(lidoService, 'getWithdrawalCredentials')
+      .mockImplementation(async () => GOOD_WC);
+  });
+
+  describe('node checks', () => {
+    it('should be on correct network', async () => {
+      const chainId = await providerService.getChainId();
+      expect(chainId).toBe(CHAIN_ID);
+    });
+
+    it('should be able to create new blocks', async () => {
+      const isMining = await providerService.provider.send('eth_mining', []);
+      expect(isMining).toBe(true);
+    });
+
+    it('should be on correct block number', async () => {
+      const provider = providerService.provider;
+      const block = await provider.getBlock('latest');
+      expect(block.number).toBe(FORK_BLOCK + 2);
+    });
+
+    it('testing address should have some eth', async () => {
+      const provider = providerService.provider;
+      const balance = await provider.getBalance(walletService.address);
+      expect(balance.gte(WeiPerEther.mul(34))).toBe(true);
+    });
+  });
+
+  describe('node operator deposit frontrun', () => {
+    it('main attack scenario', async () => {
+      // Check if the service is ok and ready to go
+      await guardianService.handleNewBlock();
+
+      const badDepositMessage = {
+        pubkey: pk,
+        withdrawalCredentials: fromHexString(BAD_WC),
+        amount: 1000000000, // gwei!
+      };
+      const badSigningRoot = computeRoot(badDepositMessage);
+      const badSig = sk.sign(badSigningRoot).toBytes();
+
+      const badDepositData = {
+        ...badDepositMessage,
+        signature: badSig,
+      };
+      const badDepositRoot = DepositData.hashTreeRoot(badDepositData);
+
+      if (!process.env.WALLET_PRIVATE_KEY) {
+        process.exit();
+      }
+      const wallet = new ethers.Wallet(process.env.WALLET_PRIVATE_KEY);
+
+      // Make a bad deposit
+      const signer = wallet.connect(providerService.provider);
+      const depositContract = DepositAbi__factory.connect(
+        DEPOSIT_CONTRACT,
+        signer,
+      );
+      await depositContract.deposit(
+        badDepositData.pubkey,
+        badDepositData.withdrawalCredentials,
+        badDepositData.signature,
+        badDepositRoot,
+        { value: ethers.constants.WeiPerEther.mul(1) },
+      );
+
+      // Mock Keys API again on new block
+      const newBlock = await providerService.provider.getBlock('latest');
+      mockKeysApi(newBlock, keysApiService);
+
+      // Pause deposits
+      await guardianService.handleNewBlock();
+
+      // Wait for confirmation
+      await new Promise((res) => setTimeout(res, 1000));
+
+      // Check if on pause now
+      const routerContract = StakingRouterAbi__factory.connect(
+        STAKING_ROUTER,
+        providerService.provider,
+      );
+      const isOnPause = await routerContract.getStakingModuleIsDepositsPaused(
+        1,
+      );
```

Comment:
We should check that the module has not been paused before the deposit

Reply:
Oh yeah, good point! Adding a check